### PR TITLE
chore(register): add s3 validation

### LINKED ISSUE #252: Gradient CLI S3 Validation
It looks like it isn't until the end of the cli installer experience that we'll get a terraform driven "this s3 string doesn't look correct" error.

### PR DESCRIPTION
**Pull request by marquiswashere**
Happy Path:
Screen Shot 2022-01-05 at 11 12 41 PM

Unhappy Path:
Screen Shot 2022-01-05 at 11 15 05 PM

Closes #252

### CODE REVIEW

**Comment by marquiswashere** [comment]
I think we have to consider in AWS-land S3 buckets, credentials, and their respective API endpoints are region specific. 

Currently we are not setting the region variable unless the user selects `AWS` as their cluster's destination. I think we should always accept the user's region selection as context for "my credentials live in this region, and my bucket live in this region." 

It looks like the goal of the conditional `if aws set region` was to omit it from this call `paperspace.ClusterCreateParams`. How do we feel if we simply move that logic "down" because of the prior statements?

File: pkg/cli/commands/clusters/register.go [hunk 1]
```diff
@@ -110,6 +132,20 @@ func ClusterRegister(client *paperspace.Client, createFilePath string) (string,
...
+			region,
+			artifactsBucketPathPrompt.Value,
+		)
+
+		if err != nil {
+			return "",
+				errors.New(
+					fmt.Sprintf("Unable to validate your S3 bucket %s from the specified credentials.", artifactsBucketPathPrompt.Value))
+		} else {
+			println(fmt.Sprintf("S3 Bucket [%s]a validated via a HeadBucket Request", artifactsBucketPathPrompt.Value))
```
Comment (paperspace-philip) [inline pkg/cli/commands/clusters/register.go:146]: I think we can just say nothing to the user, since no error means it's working

Comment (paperspace-philip) [reply pkg/cli/commands/clusters/register.go:146]: Maybe do the same at line 128 for consistency

**Comment by paperspace-philip** [comment]
> I think we have to consider in AWS-land S3 buckets, credentials, and their respective API endpoints are region specific.
> 
> Currently we are not setting the region variable unless the user selects `AWS` as their cluster's destination. I think we should always accept the user's region selection as context for "my credentials live in this region, and my bucket lives in this region."
> 
> It looks like the goal of the conditional `if aws set region` was to omit it from this call `paperspace.ClusterCreateParams`.
> 
> How do we feel if we simply move that logic "down" because of the prior statements?

I think it's a reasonable default right now. It does bring up an issue with the artifacts bucket since region is required. In general we want artifacts bucket to be s3-compatible and not tied to aws FYI. 

@sanfilip have we been manually changing the region for the artifacts if it's not us-east-1?

**Comment by sanfilip** [comment]
> 

If I understand your question: no, and this has created some problems. During the SonderMind install they (and me) were unaware that the prompt for region not allow customization of where the cluster would be installed, so this looked like a product failure to them as they picked us-central-1 and we installed in us-east-1.  I don't think we should hard-code the cluster region to us-east-1.  I believe there is a ticket for that.  I also saw the issue in the code.  We prompt for the region, but then don't use the region variable for the cluster.  For the artifacts bucket, the region should also not default without the ability to override.  Terraform state bucket and region, artifacts bucket and region, and cluster region all need to be distinct.  Hope that helps.  Let me know if I misunderstood your question.

**Comment by sanfilip** [comment]
Also, dropping down to terraform is impractical for most of these installs.  The customer would need to know ahead of time that the installer isn't going to do what they need in terms of customization.  It would have been impractical for Sondermind to make the changes to the aws eks provider resource even with their knowledge of terraform.  We would have to do it for them to make sure it would work.

**Comment by marquiswashere** [comment]
I think we're all on the same page 

Can we use the current region prompt to inform the installer of the credentials and artifact bucket?
With this approach before the `paperspace.ClusterCreateParams` function call we can simply reset the region back to an empty string or nil if it's a non-AWS installation for now? @paperspace-philip @sanfilip 

In another PR, we can decide if we want to provide a new prompt to pass in the terraform state backend/where the kube cluster will get stood up? My vote is the state backend and the infra get stood up in the same region (for AWS). 

Want to get both of your thoughts so I can close this one out.

**Comment by sanfilip** [comment]
> I think we're all on the same page
> 
> Can we use the current region prompt to inform the installer of the credentials and artifact bucket? 

Seems reasonable. Why not call it Artifacts Bucket Region, or Terraform State Bucket Region accordingly?

> 
> In another PR, we can decide if we want to provide a new prompt to pass in the terraform state backend/where the kube cluster will get stood up? My vote is the state backend and the infra get stood up in the same region (for AWS).

The Terraform State Bucket Region can be totally different than were people want to put their cluster.  We saw this with customer Croesus.  So I vote to keep them separate.

**Comment by marquiswashere** [comment]
@sanfilip sounds good from me. We'll add two prompts, one for the state bucket's region and one for the cluster.

**Comment by marquiswashere** [comment]
Plan is to add more prompts in next PR, one for terraform state, one for the EKS cluster and those applicable changes need to cascade down into terraform-land.

**Review by paperspace-philip — APPROVED** [review]

**Pull request merged**